src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/sentencing/SentencingSynchronisationService.kt

```diff
@@ -31,25 +43,77 @@ class SentencingSynchronisationService(
         event.toTelemetryProperties(it.adjustmentId)
       )
     } ?: let {
-      sentencingService.createSentencingAdjustment(nomisAdjustment.toSentencingAdjustment()).also {
-        sentencingMappingService.createNomisSentencingAdjustmentSynchronisationMapping(
-          nomisAdjustmentId = event.adjustmentId,
-          nomisAdjustmentCategory = "SENTENCE",
-          adjustmentId = it.id
-        )
-        telemetryClient.trackEvent(
-          "sentence-adjustment-created-synchronisation-success",
-          event.toTelemetryProperties(it.id),
-        )
+      sentencingService.createSentencingAdjustment(nomisAdjustment.toSentencingAdjustment()).also { adjustment ->
+        tryToCreateMapping(event, adjustment.id).also { result ->
+          telemetryClient.trackEvent(
+            "sentence-adjustment-created-synchronisation-success",
+            event.toTelemetryProperties(adjustment.id, result == MAPPING_FAILED),
+          )
+        }
       }
     }
   }
+
+  enum class MappingResponse {
+    MAPPING_CREATED,
+    MAPPING_FAILED
+  }
+
+  suspend fun tryToCreateMapping(
```

Comment:
main change here, send ourselves a message if it fails so it eventually succeeds